compute_ece.py

```diff
@@ -0,0 +1,136 @@
+        bin_corrects = np.zeros(len(bin_centers))
+        bin_confidence = np.zeros(len(bin_centers))
+
+        min_idx = self.num_classes
+        if min(bin_idxs) < min_idx:
+            min_idx = min(bin_idxs)
+        
+        for nbin in range(len(bin_centers)):
+            bin_counts[nbin] = np.sum((bin_idxs==nbin).astype(int))
+            bin_corrects[nbin] = np.sum(correct_vec[bin_idxs==nbin])
+            bin_confidence[nbin] = np.mean(expanded_preds[bin_idxs==nbin])
+
+        have_data = bin_counts > 0  
+        bin_acc = bin_corrects[have_data] / bin_counts[have_data]
+
+        ece = 0
+        for i in range(len(bin_confidence)):
+            ece += np.absolute(bin_confidence[i]-bin_acc[i]) * bin_counts[i]/np.sum(bin_counts)
+
+        return ece, bin_centers[have_data], bin_acc
+
+def get_one_hot(targets, nb_classes):
+    res = np.eye(nb_classes)[np.array(targets).reshape(-1)]
+    return res.reshape(list(targets.shape)+[nb_classes])
+
+def evaluate_ece(model, test_loader):
+    ece_criterion = ECELoss(bin_step=0.1).to(DEVICE)
+    probs_list, labels_list = [], []
+    with torch.no_grad():
+        for data in tqdm(test_loader):
+            X, y = data
+            X, y = X.to(DEVICE), y.to(DEVICE)
+            _, probs = model.sample_predict(X)
+            probs_list.append(probs)
+            labels_list.append(y)
+        probs = torch.cat(probs_list)
+        labels = torch.cat(labels_list)
+
+        probs = probs.cpu().detach().numpy()
+        labels = labels.cpu().detach().numpy()
+    
+    return ece_criterion(probs, labels)
+
+def main():
+    np.random.seed(1)
+    config = ClassConfig
+    test_ds = create_data_class(train=False, batch_size=config.batch_size, shuffle=False)
+
+    params = {
+        'lr': config.lr,
+        'hidden_units': config.hidden_units,
+        'mode': config.mode,
+        'batch_size': config.batch_size,
+        'epochs': config.epochs,
+        'x_shape': config.x_shape,
+        'classes': config.classes,
+        'num_batches': 0,
+        'train_samples': config.train_samples,
+        'test_samples': 5,
+        'mu_init': config.mu_init,
+        'rho_init': config.rho_init,
+        'prior_init': config.prior_init,
+        'mixture_prior':config.mixture_prior,
+        'save_dir': config.save_dir,
+    }
+
+    models = {
+        'BNN': BNN_Classification('bnn_classification', {**params, 'local_reparam': False, 'dropout': False}),
+        'BNN-LR': BNN_Classification('bnn_classification_lr', {**params, 'local_reparam': True, 'dropout': False}),
+        'MLP': MLP_Classification('mlp_classification', {**params, 'dropout': False}),
+        # 'dropout_class': MLP_Classification('dropout_classification', {**params, 'dropout': True}),
+        'MC-Dropout': MCDropout_Classification('mcdropout_classification', {**params, 'dropout': True}),
+        }
+
+    for _, model in models.items():
+        model.net.load_state_dict(torch.load(model.save_model_path))
```

Comment:
Could this be replaced with the functions in load_model_utils instead?

Reply:
That's what I thought to do, but we need functions e.g. predict() and enable_dropout() that are only available in the class_task objects and not the models.